From 2a109f83fc81c4ccb75cf875d11d7af62b778653 Mon Sep 17 00:00:00 2001
From: Daniel Kahn Gillmor <dkg@fifthhorseman.net>
Date: Thu, 7 Sep 2017 18:41:10 -0400
Subject: [PATCH] gpg: default El Gamal to 3072-bit keys.

* g10/keygen.c (gen_elg): update default from 2048 to 3072.

--

3072-bit El Gamal is widely considered to be 128-bit-equivalent security.

While using El Gamal might be ill-advised, if you're going to use it,
it might as well be a reasonable strength.

Signed-off-by: Daniel Kahn Gillmor <dkg@fifthhorseman.net>

(cherry picked from commit 909fbca19678e6e36968607e8a2348381da39d8c)
---
 g10/keygen.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/g10/keygen.c b/g10/keygen.c
index b7e2bf8df..745441c48 100644
--- a/g10/keygen.c
+++ b/g10/keygen.c
@@ -1752,7 +1752,7 @@ gen_elg (int algo, unsigned int nbits, KBNODE pub_root,
 
   if (nbits < 1024)
     {
-      nbits = 2048;
+      nbits = 3072;
       log_info (_("keysize invalid; using %u bits\n"), nbits );
     }
   else if (nbits > 4096)
